File: level1/K번째수/javascript/20221025/solution.test.js
import solution, { sliceAndSort } from './solution';

const array = [1, 5, 2, 6, 3, 7, 4];
const commands = [
  [2, 5, 3],
  [4, 4, 1],
  [1, 7, 3],
];

test('배열을 짜르고 정렬하자', () => {
  expect(sliceAndSort(array, commands[0])).toBe(5);
  expect(sliceAndSort(array, commands[1])).toBe(6);
  expect(sliceAndSort(array, commands[2])).toBe(3);
});

test('sliceAndSort 연산 결과를 배열에 담자', () => {
  expect(solution(array, commands)).toStrictEqual([5, 6, 3]);
});

File: programmers/level1/K번째수/java/20221024-api/test/SolutionTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    @Test
    void sliceAndSort() {
        Solution solution = new Solution();

        int[] array = {1, 5, 2, 6, 3, 7, 4};
        int[][] command = {{2, 5, 3},{4, 4, 1},{1, 7, 3}};

        assertEquals(5, solution.withApi(array, command[0]));
        assertEquals(6, solution.withApi(array, command[1]));
        assertEquals(3, solution.withApi(array, command[2]));
    }


    @Test
    void command() {
        Solution solution = new Solution();

        int[] array = {1, 5, 2, 6, 3, 7, 4};
        int[][] commands = {{2, 5, 3},{4, 4, 1},{1, 7, 3}};

        assertArrayEquals(new int[]{5, 6, 3}, solution.getResult(array, commands));
    }
}
